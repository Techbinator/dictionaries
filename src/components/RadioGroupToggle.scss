@import "../config/variables.scss";
.radio-group-container {
  input {
    display: none;
  }

  label {
    cursor: pointer;
    padding-right: 1%;
    position: relative;
    display: block;
    font-size: 18px;
    padding: 1% 0;
    border-bottom: 1px solid #ccc;
    margin-bottom: 5px;
  }

  input[type="checkbox"],
  input[type="radio"] {
    position: absolute;
    visibility: hidden !important;
  }

  input[type="checkbox"] + label,
  input[type="radio"] + label {
    &:before,
    &:after {
      content: "";
      position: absolute;
      top: 50%;
      margin-top: -7.5px;
      box-sizing: border-box;
    }

    &:before {
      width: 30px;
      height: 15px;
      right: 0px;
      background: #fff;
      border: 1px solid #e4e3e1;
      border-radius: 15px;
    }

    &:after {
      width: 15px;
      height: 15px;
      right: 15px;
      background: $primary-color;
      border-radius: 50%;
      transition: all 200ms ease-out;
    }
  }

  input[type="checkbox"]:checked + label,
  input[type="radio"]:checked + label {
    color: $success-color;
    &:after {
      right: 0px;
      background: $success-color;
    }
  }
}
