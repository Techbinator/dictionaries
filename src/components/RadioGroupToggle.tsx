import * as React from "react";
import "./RadioGroupToggle.scss";
import { IDictionaries, TactivateDictionary } from "../store/Dictionaries.type";

export interface IRadioToggleProps {
  dictionaries: IDictionaries;
  activeDictionary: string;
  activateDictionary: TactivateDictionary;
}

export default class RadioToggle extends React.Component<IRadioToggleProps> {
  public render() {
    const { dictionaries, activeDictionary, activateDictionary } = this.props;

    return (
      <div className="radio-group-container">
        <input
          type="radio"
          name="dictionary-selection"
          id="radio-0"
          checked={activeDictionary == ""}
          readOnly
        />
        <label onClick={() => activateDictionary("")} htmlFor="radio-0">
          Initial data
        </label>
        {Object.keys(dictionaries).map(key => (
          <span key={dictionaries[key].uuid}>
            <input
              name="dictionary-selection"
              type="radio"
              id={`radio-${dictionaries[key].uuid}`}
              checked={dictionaries[key].uuid === activeDictionary}
              readOnly
            />
            <label
              htmlFor={dictionaries[key].uuid}
              onClick={() => activateDictionary(dictionaries[key].uuid)}
            >
              {dictionaries[key].name}
            </label>
          </span>
        ))}
      </div>
    );
  }
}
