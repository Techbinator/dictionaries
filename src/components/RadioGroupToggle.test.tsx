import React from "react";
import RadioGroupToggle from "./RadioGroupToggle";
import { shallow } from "enzyme";
import contextMock from "../helpers/__mocks__/context.mock";

describe("RadioGroupToggle component", () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(
      <RadioGroupToggle
        dictionaries={contextMock.dictionaries}
        activeDictionary={contextMock.activeDictionary}
        activateDictionary={contextMock.activateDictionary}
      />
    );
  });
  it("should render snapshot with default values", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should select the new radio value on label click", () => {
    const selectedUUID = "619ce930-4116-11e9-9306-ef359c18301d";
    wrapper.find(`label[htmlFor="${selectedUUID}"]`).simulate("click");
    expect(contextMock.activateDictionary).toBeCalledWith(selectedUUID);
    expect(wrapper.find(`#radio-${selectedUUID}`).props().checked).toBeTruthy();
    wrapper.find(`#radio-${selectedUUID}`).simulate("change");
  });

  it("should select the default no dictionary on click", () => {
    wrapper.find(`label[htmlFor="radio-0"]`).simulate("click");
    expect(contextMock.activateDictionary).toBeCalledWith("");
  });
});
