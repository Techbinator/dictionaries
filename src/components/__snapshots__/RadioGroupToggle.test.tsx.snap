// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupToggle component should render snapshot with default values 1`] = `
<div
  className="radio-group-container"
>
  <input
    checked={false}
    id="radio-0"
    name="dictionary-selection"
    readOnly={true}
    type="radio"
  />
  <label
    htmlFor="radio-0"
    onClick={[Function]}
  >
    Initial data
  </label>
  <span
    key="619ce930-4116-11e9-9306-ef359c18301d"
  >
    <input
      checked={true}
      id="radio-619ce930-4116-11e9-9306-ef359c18301d"
      name="dictionary-selection"
      readOnly={true}
      type="radio"
    />
    <label
      htmlFor="619ce930-4116-11e9-9306-ef359c18301d"
      onClick={[Function]}
    >
      dictionary2
    </label>
  </span>
  <span
    key="4aa2cce0-4116-11e9-9306-ef359c18301d"
  >
    <input
      checked={false}
      id="radio-4aa2cce0-4116-11e9-9306-ef359c18301d"
      name="dictionary-selection"
      readOnly={true}
      type="radio"
    />
    <label
      htmlFor="4aa2cce0-4116-11e9-9306-ef359c18301d"
      onClick={[Function]}
    >
      test
    </label>
  </span>
</div>
`;
